.github/workflows/emacs-31.yml

```diff
@@ -35,14 +36,25 @@ jobs:
       HOMEBREW_EMACS_PLUS_MODE: local
 
     steps:
-      - uses: actions/checkout@v4
+      - name: Set up Homebrew
+        id: set-up-homebrew
+        uses: Homebrew/actions/setup-homebrew@master
+        with:
+          core: false
+          cask: true
+          test-bot: false
 
+      - name: Enable debug mode
+        run: |
+          echo 'HOMEBREW_DEBUG=1' >> "${GITHUB_ENV}"
+          echo 'HOMEBREW_VERBOSE=1' >> "${GITHUB_ENV}"
+        if: runner.debug
       - name: Install xquartz
         if: contains(matrix.build_opts, '--with-x11')
         run: brew install --cask xquartz
 
       - name: Build emacs-plus@31 ${{ matrix.build_opts }}
-        run: brew install ./Formula/emacs-plus@31.rb ${{ matrix.build_opts }} --verbose
+        run: brew install emacs-plus@31 ${{ matrix.build_opts }} --verbose
```

Comment:
Why?

Reply:
Running` brew install ./Formula/emacs-plus@31.rb --with-mps`  thrown error :  

`Error: invalid option: --with-mps`.   [Link](https://github.com/d12frosted/homebrew-emacs-plus/actions/runs/10323706429/job/28581651325#step:4:147)

While same content  place in `/opt/homebrew/Library/Taps/d12frosted/homebrew-emacs-plus/Formula`

Run `brew install emacs-plus@31 --with-mps` work fine

Reply:
That's strange. I wonder how other options work. I don't remember any details 😂  Need to investigate.

----

*update:*

So I tried adding a new argument in [a separate PR](https://github.com/d12frosted/homebrew-emacs-plus/pull/708) and it seems that CI [picked it up](https://github.com/d12frosted/homebrew-emacs-plus/actions/runs/10465548134/job/28980967099?pr=708#step:4:746) without this change. 

I don't remember details, but it's not really clear to me what exactly is being built when `emacs-plus@31` is referenced instead of specific file.

I also don't really understand why do we need [Setup Homebrew Github Action](https://github.com/Homebrew/actions/tree/master/setup-homebrew). Standard GitHub macos runners have brew installed and configured out of the box. 

> I pulled the workflow from Homebrew-core and it works great.

I think they need it because they are running on ubuntu image which doesn't have homebrew ootb.

Reply:
That's weird... if the simplest case works fine, maybe the problem is elsewhere. 

> I also don't really understand why do we need Setup Homebrew Github Action. Standard GitHub macos runners have brew installed and configured out of the box.

I wan to make sure we're testing this formula like a normal user would use brew. but it seems to be bringing in more complexity than necessary.